contracts/utilities/TokenFeeCollector.sol

```diff
@@ -0,0 +1,150 @@
+        );
+        TokenInfo storage tInfo = tokenPayees[tokenAddress];
+        for(uint i = 0; i < tInfo.payees.length; i++) {
+            require(tInfo.payees[i] != devAddress, "This devAddress is already registered");
+        }
+        tInfo.payees.push(devAddress);
+        tInfo.paymentBips.push(paymentBips);
+    }
+
+    function removePayee(address tokenAddress, address devAddress) external onlyOwner {
+        require(
+            tokenPayees[tokenAddress].isValue,
+            "This token is not registered, use registerToken to initialize it"
+        );
+        TokenInfo storage tInfo = tokenPayees[tokenAddress];
+        uint payeeIndex = tInfo.payees.length+1;
+        for(uint i = 0; i < tInfo.payees.length; i++) {
+            if (tInfo.payees[i] == devAddress) {
+                payeeIndex = i;
+                break;
+            }
+        }
+        require(payeeIndex < tInfo.payees.length, "Payee not found");
+        _removeElement(tokenAddress, payeeIndex);
+    }
+
+    function editPayee(address tokenAddress, address devAddress, uint paymentBips) external onlyOwner {
+        require(
+            tokenPayees[tokenAddress].isValue,
+            "This token is not registered, use registerToken to initialize it"
+        );
+        TokenInfo storage tInfo = tokenPayees[tokenAddress];
+        uint payeeIndex = tInfo.payees.length+1;
+        for(uint i = 0; i < tInfo.payees.length; i++) {
+            if (tInfo.payees[i] == devAddress) {
+                payeeIndex = i;
+                break;
+            }
+        }
+        require(payeeIndex < tInfo.payees.length, "Payee not found");
+        tInfo.paymentBips[payeeIndex] = paymentBips;
+    }
+
+    function viewPayee(address tokenAddress, uint index) external view onlyOwner returns(address, uint) {
+        require(
+            tokenPayees[tokenAddress].isValue,
+            "The token is not registered, use registerToken to initialize it"
+        );
+        TokenInfo storage tInfo = tokenPayees[tokenAddress];
+        require(index < tInfo.payees.length, "You're reading from outside the boundaries");
+        return (tInfo.payees[index], tInfo.paymentBips[index]);
+    }
+
+    function collectFee(address tokenAddress) external {
+        TokenInfo storage tInfo = tokenPayees[tokenAddress];
+        uint payeeIndex = tInfo.payees.length+1;
+        for(uint i = 0; i < tInfo.payees.length; i++) {
+            if (tInfo.payees[i] == msg.sender) {
+                payeeIndex = i;
+                break;
+            }
+        }
+        require(payeeIndex < tInfo.payees.length || msg.sender == owner(), "You're not a payee or the owner, don't collect");
+        uint totalBips = 0;
+        for(uint i = 0; i < tInfo.payees.length; i++) {
+            totalBips += tInfo.paymentBips[i];
+        }
+
+        require(totalBips <= maxBips, "The total payout configured exceeds 100%, contact the feeCollect owner");
```

Comment:
check on totalBips should be enforced somewhere else. For example, when we set and update paymentBips instead of during collection